darwin/dataset/download_manager.py

```diff
@@ -292,14 +297,39 @@ def _download_image(url: str, path: Path, api_key: str) -> None:
             response = requests.get(url, headers={"Authorization": f"ApiKey {api_key}"}, stream=True)
         # Correct status: download image
         if response.ok:
-            with open(str(path), "wb") as file:
-                for chunk in response:
-                    file.write(chunk)
+            if has_json_content_type(response):
+                _fetch_multiple_files(path, response)
+            else:
+                _write_file(path, response)
             return
         # Fatal-error status: fail
         if 400 <= response.status_code <= 499:
-            raise Exception(f"Request to ({url}) failed. Status code: {response.status_code}, content:\n{get_response_content(response)}.")
+            error = f"Request to ({url}) failed. Status code: {response.status_code}, content:\n{get_response_content(response)}."
+            print(error)
```

Comment:
For some reason exceptions raised from _download_image are not displayed, do you know what is the reason?

Reply:
The reason is that we don't allow exceptions to be visible to user, this is not pretty. If you want to inform user about something, most of the code seems to be using `cli_functions._error` to print something human readable. Other details should be logged instead.

Reply:
@rslota Is it a way for user to enable exception displaying for debugging?

Reply:
I honestly don't know. @Fl4m3Ph03n1x ? There should be, if there isn't now, we definitely need to make this easy while rewriting.

Reply:
@rslota there seems to be a confusion here. We raise exceptions in lower level code. We do not print. Ever. The higher level functions then catch these exceptions and print a human readable message. An example of this can be seen when creating a dataset:

```
def create_dataset(dataset_slug: str) -> None:
    """
    Creates a dataset remotely. Exits the application if the dataset's name is already taken or is
    not valid.

    Parameters
    ----------
    dataset_slug : str
        Slug of the new dataset.
    """
    identifier: DatasetIdentifier = DatasetIdentifier.parse(dataset_slug)
    client: Client = _load_client(team_slug=identifier.team_slug)
    try:
        dataset: RemoteDataset = client.create_dataset(name=identifier.dataset_slug)
        print(
            f"Dataset '{dataset.name}' ({dataset.team}/{dataset.slug}) has been created.\nAccess at {dataset.remote_path}"
        )
        print_new_version_info(client)
    except NameTaken:
        _error(f"Dataset name '{identifier.dataset_slug}' is already taken.")
    except ValidationError:
        _error(f"Dataset name '{identifier.dataset_slug}' is not valid.")
```

This is a high level function that catches exceptions from bellow and then uses a Rich theme to print a Human readable error. 
If no message is ever displayed when downloading images, then we need to find out which high level function is using this one and add the exception there. 

We dont show full out blown exceptions to users, when that happens, something failed. Yet, it would still be preferable to showing nothing.

Reply:
I know, this all makes sense, I was asking if there is a way, a debug mode of sorts to show exceptions without changing code for debugging.